botorch/posteriors/multitask.py

```diff
@@ -226,9 +227,26 @@ def rsample_from_base_samples(
             train_diff.reshape(*train_diff.shape[:-2], -1) - updated_obs_samples
         )
         train_covar_plus_noise = self.train_train_covar + self.train_noise
-        obs_solve = train_covar_plus_noise.solve(obs_minus_samples.unsqueeze(-1))
+
+        # permute dimensions to move largest batch dimension to the end (more efficient 
```

Comment:
Can you move this permute-solve-unpermute logic into a `_permute_solve()` helper so that the code is easier to read as a whole?